### PR TITLE
Small UI fixes

### PR DESCRIPTION
**Pull request by philipbraatz**
Screenshot path .ToString("s") had invalid characters for me, so it couldn't save. I don't know what format you would like it in but currently using MM-dd-yyyy HH-mm-ss.

Handling bad game states and logging them to prevent crashing.

OptionMenu.cs | clear children stops the duplication of GUI rows but removes the discriptive text :/ Descriptive text should be declared with the rest of the menu options I would think.

I have created buttons to reorder the servers BUT I have no clue how things link up in this project and would like a hand :)

### CODE REVIEW

File: src/Alex/Gamestates/MainMenu/OptionsState.cs [hunk 1]
```diff
@@ -14,6 +14,8 @@ public OptionsState(GuiPanoramaSkyBox skyBox) : base(skyBox)
 
         protected override void OnInit(IGuiRenderer renderer)
         {
+            Body.ClearChildren();
```
Comment (kennyvv) [inline src/Alex/Gamestates/MainMenu/OptionsState.cs:17]: Instead of clearing we should probably just move the element creation to the constructor.

File: src/Alex/Utils/Texture2DJsonConverter.cs [hunk 2]
```diff
@@ -56,6 +56,10 @@ public override void WriteJson(JsonWriter writer, Texture2D value, JsonSerialize
 		{
 			try
 			{
+				if (reader.Value == null)
```
Comment (kennyvv) [inline src/Alex/Utils/Texture2DJsonConverter.cs:59]: Has this ever occured? 0.o

File: src/Alex/Gamestates/MainMenu/Options/OptionsStateBase.cs [hunk 3]
```diff
@@ -118,8 +118,18 @@ protected override void OnInit(IGuiRenderer renderer)
 
                 if (canConstruct)
                 {
-                    state = (TGameState) constructor.Invoke(passedParameters);
-                    break;
+                    try
```
Comment (kennyvv) [inline src/Alex/Gamestates/MainMenu/Options/OptionsStateBase.cs:121]: I think we should just fix the error instead of catching it honestly, it's alright if it crashes here i feel. As long as it doesn't crash mid-game.

File: src/Alex.Networking.Java/NetConnection.cs [hunk 4]
```diff
@@ -578,6 +578,9 @@ private bool SocketConnected(Socket s)
         {
 	        try
 	        {
+				if (s == null)
```
Comment (kennyvv) [inline src/Alex.Networking.Java/NetConnection.cs]: Could you align this please? :P

**Review by kennyvv — COMMENTED** [review]
First of all, thank you for diving into the code and creating your first Pull Request! I hope to see many more to come :)

Just a few small comments. Thanks again for your help!

File: src/Alex/Utils/Texture2DJsonConverter.cs (hunk 2 above)
Comment (philipbraatz) [reply src/Alex/Utils/Texture2DJsonConverter.cs:59]: Yes, seems to happen 4 times in a row when you click muliplayer. I would rather not parse them when there is nothing to read, but the catch is handling it currently. Reader.Path = [0].CachedIcon, [1].CachedIcon and so on

File: src/Alex.Networking.Java/NetConnection.cs [hunk 5]
```diff
@@ -26,633 +26,636 @@ namespace Alex.Networking.Java
     public class NetConnection : IDisposable
```
Comment (philipbraatz) [inline src/Alex.Networking.Java/NetConnection.cs:26]: Oh lord, I guess the formatting on this file was slightly wack!

**Comment by philipbraatz** [comment]
@kennyvv is there anything else I need to do to get this approved?

File: src/Alex.Networking.Java/NetConnection.cs (hunk 5 above)
Comment (kennyvv) [reply src/Alex.Networking.Java/NetConnection.cs:26]: This needs to be removed from your PR, they should only contain changes required for your pull request.

File: src/Alex/Entities/PlayerController.cs [hunk 6]
```diff
@@ -187,7 +187,7 @@ private void OnEndDraw(object? sender, EventArgs e)
 					Graphics.GetBackBufferData(data);
 
 					var pics = Environment.GetFolderPath(Environment.SpecialFolder.MyPictures);
-					var screenshotPath = Path.Combine(pics, $"alex-{DateTime.Now.ToString("s")}.png");
+					var screenshotPath = Path.Combine(pics, $"alex-{DateTime.Now.ToString("MM-dd-yyyy HH-mm-ss")}.png");
```
Comment (kennyvv) [inline src/Alex/Entities/PlayerController.cs:190]: This really should have been a different PR ;)
Try making sure you only commit changes related to the original scope of the PR

File: src/Alex/Gamestates/MainMenu/Options/OptionsStateBase.cs (hunk 3 above)
Comment (kennyvv) [reply src/Alex/Gamestates/MainMenu/Options/OptionsStateBase.cs:121]: Please revert this.

File: src/Alex/Gamestates/MainMenu/OptionsState.cs (hunk 1 above)
Comment (kennyvv) [reply src/Alex/Gamestates/MainMenu/OptionsState.cs:17]: See my previous comment.

File: src/Alex/Gamestates/Multiplayer/GuiServerListEntryElement.cs [hunk 7]
```diff
@@ -44,7 +45,7 @@ public IPEndPoint ConnectionEndpoint
 			}
 		}
 
-		private readonly TextureElement     _serverIcon;
+		private readonly GuiMoveServerElement _serverIcon;
```
Comment (kennyvv) [inline src/Alex/Gamestates/Multiplayer/GuiServerListEntryElement.cs:48]: Without the logic behind the MoveServer stuff, this should probably be reverted or implemented.

File: src/Alex/Gamestates/Multiplayer/MultiplayerServerSelectionState.cs [hunk 8]
```diff
@@ -25,68 +25,68 @@ namespace Alex.Gamestates.Multiplayer
 {
     public class MultiplayerServerSelectionState : ListSelectionStateBase<GuiServerListEntryElement>
```
Comment (kennyvv) [inline src/Alex/Gamestates/Multiplayer/MultiplayerServerSelectionState.cs:26]: I might be mistaken, but i think this is only a reformatting? If so, please revert :)

If there are other changes, please fix the formatting so i can actually see what changed.

File: src/Alex/Graphics/Models/Items/ItemModelRenderer.cs [hunk 9]
```diff
@@ -45,7 +45,7 @@ public override bool Cache(ResourceManager pack)
 
             if (pack.TryGetBitmap(texture, out var bitmap))
             {
-                if (ResourcePackModel.ParentName.Path.Contains("handheld", StringComparison.InvariantCultureIgnoreCase))
+                if (ResourcePackModel.ParentName?.Path.Contains("handheld", StringComparison.InvariantCultureIgnoreCase) ?? false)
```
Comment (kennyvv) [inline src/Alex/Graphics/Models/Items/ItemModelRenderer.cs:48]: Could you explain to me why you did this?

File: src/Alex/Gui/Elements/GuiMoveServerElement.cs [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+                Width = 20,
+                Height = 10,
+                MinHeight = 10,
+                MinWidth = 20,
+                MaxWidth = 20,
+                MaxHeight = 10,
+                Margin = Thickness.One,
+            };
+
+            Children.Add(JoinServerButton);
```
Comment (kennyvv) [inline src/Alex/Gui/Elements/GuiMoveServerElement.cs:64]: You should use AddChild instead

File: src/Alex/ResourceManager.cs [hunk 11]
```diff
@@ -166,22 +166,29 @@ public bool TryLoadResourcePackInfo(string file, out ResourcePackManifest[] mani
...
+                            {
+                                manifests[i].Name = archive.Name;
+                            }
                         }
                     }
                 }
             }
+            catch(Exception e)
+            {
+                Log.Error($"Could not load resource pack: {file}",e);
```
Comment (kennyvv) [inline src/Alex/ResourceManager.cs:190]: This seems a bit outside of the scope of this PR.
+ Formatting

Formatting changes throughout a file (or multiple files for that matter) are very annoying to review. Sorry!

File: src/Alex/Utils/Texture2DJsonConverter.cs (hunk 2 above)
Comment (kennyvv) [reply src/Alex/Utils/Texture2DJsonConverter.cs:59]: Fair enough, tho we should probably check if the `existingValue` param is non-null and return that instead of null :)

**Review by kennyvv — CHANGES_REQUESTED** [review]

**Comment by kennyvv** [comment]
Thanks for contributing! 
I hope i didn't come across as rude in my review, but try to stick to specific features when opening a PR.
So for example, the move servers in the serverlist feature. This is a great idea! It'd be even better if this PR was specifically doing that and not some other misc stuff.

That way it's also a lot easier for me to review and merge :)

Thanks again!

**Comment by philipbraatz** [comment]
Thank you for reviewing. With this feedback, I will split this up in more targeted pull requests
